packages/@lwc/engine-dom/src/custom-elements/create-scoped-registry.ts

```diff
@@ -42,24 +56,16 @@ export function createScopedRegistry(): CreateScopedConstructor {
     const pendingRegistryForElement = new WeakMap<HTMLElement, Definition>();
     const definitionForConstructor = new WeakMap<CustomElementConstructor, Definition>();
     const registeredUserCtors = new WeakSet<CustomElementConstructor>();
+    const registeredPivotCtors = new WeakSet<CustomElementConstructor>();
 
     const pivotCtorByTag = new Map<string, CustomElementConstructor>();
     const globalDefinitionsByTag = new Map<string, Definition>();
     const globalDefinitionsByClass = new Map<CustomElementConstructor, Definition>();
     const awaitingUpgrade = new Map<string, Set<HTMLElement>>();
+    const pendingWhenDefinedCallbacks = new Map<string, WhenDefinedCallback[]>();
```

Comment:
I considered keeping track of the `Promise` as well as the `resolve` function, so that we didn't need an array of callbacks and could just keep returning the same `Promise`, but it got really hairy to deal with the Promise constructor callbacks. An array was a lot simpler.